src/core/Button/Button.tsx

```diff
@@ -127,12 +123,8 @@ class BaseButton extends Component<ButtonProps> {
   }
 }
 
-const StyledButton = styled(
-  ({ tokens, ...passProps }: ButtonProps & InternalTokensProp) => (
-    <BaseButton {...passProps} />
-  ),
-)`
-  ${(props) => baseStyles(props)}
+const StyledButton = styled((props: ButtonProps) => <BaseButton {...props} />)`
+  ${baseStyles}
```

Comment:
In this and in similar places it seems to be possible to use simplified version as the props are not used like those were before.

Like this:
```
const StyledButton = styled(BaseButton)`
  ${baseStyles}
`;
```

Reply:
Good point here as well. Not sure if there are any perf implications, but it's better to have it cleaner.

Reply:
These should be now better. Did some refactoring in few places and moved AutoId to main level. The ones with className logic in styled were left untouched.